src/ble/tests/TestBleLayer.cpp

```diff
@@ -383,5 +408,127 @@ TEST_F(TestBleLayer, ExceedBleConnectionEndPointLimit)
+}
+
+TEST_F(TestBleLayer, NewBleConnectionByDiscriminatorsNoConnectionDelegate)
+{
+    // Set up the BleLayerTestAccess accessor class to manipulate the BleConnectionDelegate of BleLayer
+    chip::Test::BleLayerTestAccess access(this);
+    access.SetConnectionDelegate(nullptr);
+
+    SetupDiscriminator discriminators[] = { SetupDiscriminator(), SetupDiscriminator() };
+    Span<const SetupDiscriminator> discriminatorsSpan(discriminators, 2);
+
+    auto OnSuccess = [](void * appState, uint16_t matchedLongDiscriminator, BLE_CONNECTION_OBJECT connObj) {};
+    auto OnError   = [](void * appState, CHIP_ERROR err) {};
+
+    EXPECT_EQ(NewBleConnectionByDiscriminators(discriminatorsSpan, this, OnSuccess, OnError), CHIP_ERROR_INCORRECT_STATE);
+}
+
+TEST_F(TestBleLayer, NewBleConnectionByDiscriminatorsNoBleTransportLayer)
+{
+    chip::Test::BleLayerTestAccess access(this);
+    access.SetConnectionDelegate(this);
+    mBleTransport = nullptr;
+
+    SetupDiscriminator discriminators[] = { SetupDiscriminator(), SetupDiscriminator() };
+    Span<const SetupDiscriminator> discriminatorsSpan(discriminators, 2);
+
+    auto OnSuccess = [](void * appState, uint16_t matchedLongDiscriminator, BLE_CONNECTION_OBJECT connObj) {};
+    auto OnError   = [](void * appState, CHIP_ERROR err) {};
+
+    EXPECT_EQ(NewBleConnectionByDiscriminators(discriminatorsSpan, this, OnSuccess, OnError), CHIP_ERROR_INCORRECT_STATE);
+}
+
+TEST_F(TestBleLayer, NewConnectionByDiscriminatorsSuccess)
+{
+    chip::Test::BleLayerTestAccess access(this);
+    access.SetConnectionDelegate(this);
+
+    SetupDiscriminator discriminators[] = { SetupDiscriminator(), SetupDiscriminator() };
+    discriminators[0].SetLongValue(1234);
+    discriminators[1].SetLongValue(3333);
+    Span<const SetupDiscriminator> discriminatorsSpan(discriminators, 2);
+
+    auto OnSuccess = [](void * appState, uint16_t matchedLongDiscriminator, BLE_CONNECTION_OBJECT connObj) {
+        BleLayer * testLayer = static_cast<BleLayer *>(appState);
+        chip::Test::BleLayerTestAccess tempAccess(testLayer);
+
+        tempAccess.CallOnConnectionComplete(appState, connObj);
+    };
+    auto OnError             = [](void * appState, CHIP_ERROR err) { FAIL() << "OnError should not be called in this test"; };
+    BleLayer * bleLayerState = this;
+
+    EXPECT_EQ(NewBleConnectionByDiscriminators(discriminatorsSpan, this, OnSuccess, OnError), CHIP_NO_ERROR);
+
+    // Simulate a successful connection by calling the success callback directly
+    OnSuccess(bleLayerState, discriminatorsSpan[0].GetLongValue(), GetConnectionObject());
+
+    // Verify that the success callback was called
+    EXPECT_EQ(mOnConnectionCompleteCount, 1);
+    EXPECT_EQ(mOnConnectionErrorCount, 0);
+}
+
+// Checks that the connection could not be established due to an error
+TEST_F(TestBleLayer, NewConnectionByDiscriminatorsError)
+{
+    chip::Test::BleLayerTestAccess access(this);
```

Comment:
Why the `chip::` prefix here and elsewhere in this PR?  Is this not all inside `namespace chip`?